### PR TITLE
add document for explain/explain analyze of mpp query.

### PR DESCRIPTION
**Pull request by windtalker**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

This pr add document for `explain/explain analyze` of mpp query.
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md
Comment (TomShawn) [inline TOC.md:105]: These documents are added to TOC in https://github.com/pingcap/docs-cn/pull/5561/files. PTAL

File: explain-mpp.md
Comment (ilovesoup) [inline explain-mpp.md:92]: Please provide an sample of join.

**Review by ilovesoup — APPROVED** [review]
LGTM except need an sample of join with shuffle.

File: TOC.md
Comment (windtalker) [reply TOC.md:105]: OK, I will delete these change in my PR, but when will that pr be merged? Looks like no one has dealt with that pr for 11 days.

File: explain-mpp.md
Comment (qiancai) [reply explain-mpp.md:92]: @windtalker, we need to complete the CN docs today. PTAL, thanks!

**Comment by windtalker** [comment]
Currently blocked by #5702 and #5561

File: explain-mpp.md
Comment (windtalker) [reply explain-mpp.md:92]: You mean this pr should be merged today? Currently it is blocked by #5702 and #5561

File: TOC.md
Comment (TomShawn) [reply TOC.md]: What about using the added TOC entries in [#5561](https://github.com/pingcap/docs-cn/pull/5561/files#diff-19c55decc75725c7053c81221f882f7e759ced8c296087bbb5dbdb90f2d365dcR105-R112) without waiting for #5561's merge? Personally, I think they are more expressive.  If you think any modification is needed, feel free to apply it.

Comment (windtalker) [reply TOC.md]: You mean not change `TOC.md` in this pr, just updating #5561 after this pr is merged?

Comment (windtalker) [reply TOC.md]: Or maybe just merge this pr, and let the owner of #5561 to resolve the conflict?

Comment (TomShawn) [reply TOC.md]: "You mean not change TOC.md in this pr, just updating #5561 after this pr is merged?"
--------------------------------------------------------------------------------------------------
Good idea! We just need to add "TiDB mpp 执行计划" in this PR and leave other things to 5561.

Comment (windtalker) [reply TOC.md]: OK, I've remove TOC related changes, PTAL

Comment (TomShawn) [reply TOC.md]: No. You still need to add this doc to TOC in this PR. Otherwise, users cannot find this doc on the PingCAP doc site. What about `MPP 模式查询的执行计划`？

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:67]:
```suggestion
上述例子中 ExchangeSender 的 ExchangeType 为 HashPartition 以及 PassThroough，分别对应于 Hash Aggregation 运算以及向 TiDB 返回数据。
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:69]:
```suggestion
另外一个典型的 MPP 应用为 join 运算。TiDB MPP 支持两种类型的 join，分别为：
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:72]: Broadcast 还是 broadcast? 请统一大小写。能否写“广播”？

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:79]: 这是一条长的 SQL 语句吗？

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:102]: 建议使用无序列表罗列，使阅读更清晰。
```suggestion
以上执行计划中，`[TableFullScan_20, Selection_21, ExchangeSender_22]` 完成表 b 的数据读取并通过 HashPartition 的方式把数据 shuffle 给上游 MPP Task。`[TableFullScan_16, Selection_17, ExchangeSender_18]` 完成表 a 的数据读取并通过 HashPartition 的方式把数据 shuffle 给上游 MPP Task。`[ExchangeReceiver_19, ExchangeReceiver_23, HashJoin_44, ExchangeSender_47]` 完成 join 并把数据返回给 TiDB。
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:74]:
```suggestion
典型的 Shuffle Hash Join 执行计划如下：
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:104]:
```suggestion
典型的 Broadcast Join 执行计划如下：
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:129]: + "小表"和"大表"分别指 a 和 b 吗？
+ 建议使用无序列表罗列
```suggestion
以上执行计划中，`[TableFullScan_17, Selection_18, ExchangeSender_19]` 从小表读数据并 broadcast 给大表数据所在的各个节点。`[TableFullScan_21, Selection_22, ExchangeReceiver_20, HashJoin_43, ExchangeSender_46]` 完成 join 并将数据返回给 TiDB。
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:131]: SQL 关键字请尽量大写
```suggestion
## 对 MPP 模式的查询使用 `EXPLAIN ANALYZE`
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:133]:
```suggestion
`EXPLAIN ANALYZE` 语句与 `EXPLAIN` 类似，但还会输出一些运行时的信息。一个简单的 `EXPLAIN ANALYZE` 输出信息如下：
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:156]:
```suggestion
与 `EXPLAIN` 的输出结果相比，ExchangeSender 的 operator info 列中多了 task id 的输出，记录了该 query fragment 实例化成的 MPP Task 的 task id。此外 MPP Operator 中都会有 `threads` 这一列，这列记录了 MPP 在执行该算子时使用的并发数（对于由多个节点组成的集群，该并发数是所有节点并发数相加的结果）。
```

File: explain-mpp.md
Comment (windtalker) [reply explain-mpp.md:82]: 前三个 SET 是设置一下相关的控制变量，因为这个 SQL 需要先设置一下相关的控制变量才能得出下面的查询计划。

File: explain-mpp.md
Comment (windtalker) [reply explain-mpp.md:75]: 可以，我改一下

File: explain-mpp.md
Comment (windtalker) [reply explain-mpp.md:22]: 有的，我改一下

File: explain-mpp.md
Comment (windtalker) [reply explain-mpp.md]: 要的，比如第一个查询片段还需要完成读表的操作

File: explain-mpp.md
Comment (windtalker) [reply explain-mpp.md]: 请问下无序列表具体指什么？怎么使用？

**Comment by windtalker** [comment]
Now this pr is blocked by #5702 , because we have an anchor to /tiflash/use-tiflash.md#使用-mpp-模式

File: explain-mpp.md
Comment (TomShawn) [reply explain-mpp.md]: 无序列表就是：

+ `[TableFullScan_20, Selection_21, ExchangeSender_22]` 完成表 b 的数据读取并通过 HashPartition 的方式把数据 shuffle 给上游 MPP Task。
+ `[TableFullScan_16, Selection_17, ExchangeSender_18]` 完成表 a 的数据读取并通过 HashPartition 的方式把数据 shuffle 给上游 MPP Task。
+ `[ExchangeReceiver_19, ExchangeReceiver_23, HashJoin_44, ExchangeSender_47]` 完成 join 并把数据返回给 TiDB。

Comment (windtalker) [reply explain-mpp.md]: 好的，已更新，PTAL

File: TOC.md
Comment (TomShawn) [inline TOC.md:105]:
```suggestion
      + [MPP 模式查询的执行计划](/explain-mpp.md)
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:24]:
```suggestion
在 MPP 模式下，一个查询在逻辑上会被切分为多个 MPP 查询片段 (query fragment)。示例如下：
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:67]:
```suggestion
* Broadcast：ExchangeSender 通过广播的方式把数据分发给上游的 MPP 任务，通常在 Broadcast Join 中使用。
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:70]:
```suggestion
上述例子中 ExchangeSender 的 ExchangeType 为 HashPartition 以及 PassThrough，分别对应于 Hash Aggregation 运算以及向 TiDB 返回数据。
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:74]:
```suggestion
* Shuffle Hash Join：join 的 input 通过 HashPartition 的方式 shuffle 数据，上游的 MPP 任务进行分区内的 join。 
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:138]:
```suggestion
* `[TableFullScan_17, Selection_18, ExchangeSender_19]` 从小表读数据并 Broadcast 给大表（表 a）数据所在的各个节点。
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:166]:
```suggestion
与 `EXPLAIN` 相比，ExchangeSender 的 `operator info` 中多了 `task id` 的输出，其记录了该查询片段实例化成的 MPP 任务的任务 ID。此外 MPP 算子中都会有 `threads` 这一列，这列记录了 MPP 在执行该算子时使用的并发数（如果集群由多个节点组成，该并发数是所有节点并发数相加的结果）。
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:8]:
```suggestion
TiDB 支持使用 [MPP 模式](/tiflash/use-tiflash.md#使用-mpp-模式)来执行查询。在 MPP 执行模式下，SQL 优化器会生成 MPP 的执行计划。注意 MPP 模式仅对有 [TiFlash](/tiflash/tiflash-overview.md) 副本的表生效。
```

File: explain-mpp.md
Comment (TomShawn) [inline explain-mpp.md:82]: SET tidb_opt_broadcast_join=0;
SET tidb_broadcast_join_threshold_count=0;
SET tidb_broadcast_join_threshold_size=0;
EXPLAIN SELECT COUNT(*) FROM t1 a JOIN t1 b ON a.id = b.id;

**Pull request merged**
